main.tf

```
@@ -16,16 +16,17 @@ data "aws_caller_identity" "current" {
 # https://registry.terraform.io/providers/DataDog/datadog/latest/docs/resources/integration_aws
 # https://docs.datadoghq.com/api/v1/aws-integration/
 resource "datadog_integration_aws" "integration" {
-  count                            = local.enabled ? 1 : 0
-  account_id                       = local.aws_account_id
-  role_name                        = module.this.id
-  filter_tags                      = var.filter_tags
-  host_tags                        = var.host_tags
-  excluded_regions                 = var.excluded_regions
-  account_specific_namespace_rules = var.account_specific_namespace_rules
-  cspm_resource_collection_enabled = var.cspm_resource_collection_enabled
-  metrics_collection_enabled       = var.metrics_collection_enabled
-  resource_collection_enabled      = var.resource_collection_enabled
+  count                                = local.enabled ? 1 : 0
+  account_id                           = local.aws_account_id
+  role_name                            = module.this.id
+  filter_tags                          = var.filter_tags
+  host_tags                            = var.host_tags
+  excluded_regions                     = var.excluded_regions
+  account_specific_namespace_rules     = var.account_specific_namespace_rules
+  cspm_resource_collection_enabled     = var.cspm_resource_collection_enabled
+  metrics_collection_enabled           = var.metrics_collection_enabled
+  resource_collection_enabled          = var.resource_collection_enabled
+  extended_resource_collection_enabled = var.extended_resource_collection_enabled
```

Preserve backward compatibility while removing deprecated resource:

```suggestion
  extended_resource_collection_enabled = var.resource_collection_enabled != null ? var.resource_collection_enabled : var.extended_resource_collection_enabled
```